### PR TITLE
Move mapTypeOutOfContext() from GenericEnvironment to TypeBase

### PR DESCRIPTION
**Pull request by slavapestov**
It no longer requires a generic environment.

### CODE REVIEW

**Comment by slavapestov** [comment]
@DougGregor Does this offend you?

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux
https://github.com/apple/swift-lldb/pull/257

**Comment by slavapestov** [comment]
@swift-ci Please smoke test
apple/swift-lldb#257

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux
apple/swift-lldb#257

**Comment by slavapestov** [comment]
@swift-ci Please smoke test
apple/swift-lldb#257

**Comment by slavapestov** [comment]
apple/swift-lldb#257
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
apple/swift-lldb#257
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
apple/swift-lldb#257
@swift-ci Please smoke test Linux

File: lib/AST/GenericEnvironment.cpp [hunk 1]
```diff
@@ -120,25 +120,13 @@ Type GenericEnvironment::mapTypeIntoContext(GenericEnvironment *env,
...
-Type GenericEnvironment::mapTypeOutOfContext(Type type) const {
-  type = type.subst([&](SubstitutableType *t) -> Type {
-                      return cast<ArchetypeType>(t)->getInterfaceType();
-                    },
-                    MakeAbstractConformanceForGenericType(),
-                    SubstFlags::AllowLoweredTypes);
-  assert(!type->hasArchetype() && "not fully substituted");
-  return type;
+Type TypeBase::mapTypeOutOfContext() {
+  assert(!hasTypeParameter() && "already have an interface type");
```
Comment (DougGregor) [inline lib/AST/GenericEnvironment.cpp]: Shouldn't we have a fast path for `!hasArchetype()`?

**Comment by DougGregor** [comment]
Nah, doesn't offend me. It's annoying to have to drag a context around for no reason.

**Pull request merged**